### PR TITLE
MM-16268 Community plugin process.

### PR DESCRIPTION
**Pull request by crspeller**
Draft proposal for process to get plugins onto community.

### CODE REVIEW

File: site/content/extend/plugins/community_process.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+- BSD 2-clause
+- BSD 3-clause
+
+[More info](https://www.apache.org/legal/resolved.html#category-a)
+
+### Complete code review
+
+A more through code review is performed before allowing a plugin on ci-extensions-release. This review works the same as the basic code review, but the developers performing the review will be more through. If the developer that performed the first review is available, they should be one of the reviewers. One of the reviewers should focus their review on the any security implications of the plugin.
+
+### QA pass
```
Comment (crspeller) [inline site/content/extend/plugins/community_process.md]: @DHaussermann Hoping you can help flesh out what QA would want to see here.

File: site/content/extend/plugins/community_process.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+QA pass involves getting a memeber of our QA team to take a look and verify the functionality advertised by your plugin.
+
+### PM/UX Review
```
Comment (crspeller) [inline site/content/extend/plugins/community_process.md]: @aaronrothschild Hoping you can flesh out what PM wants to see here.

File: site/content/extend/plugins/community_process.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+## Step Definitions
+
+### Basic Code Review
+
+Basic code review of an experimental plugin involves a quick review by a [core committer](/contribute/getting-started/core-committers/) to verify that the plugin does what it says it does and to provide any guidance and feedback. To make it easier to provide feedback, a PR can be made that contains all the code of the plugin that isn't the boilerplate from mattermost-plugin-starter-template.
+
+### CI System Setup
+
+Setting up the CI system for your plugin will allow continuous testing of your master branch and releases on our testing servers. Master branch testing is done on https://ci-extensions.azure.k8s.mattermost.com/ and release testing is done on https://ci-extensions-release.azure.k8s.mattermost.com/
```
Comment (hanzei) [inline site/content/extend/plugins/community_process.md]: Is setting up this CI possible for non mattermost plugins?

Comment (crspeller) [reply site/content/extend/plugins/community_process.md]: Yes. The next paragraph explains that you will need to setup your own CI. Then the ci-extentions servers can pull from there. 

I would love suggestions on how to make this easy for people. I was thinking about making a circleci orb, but it seems quite difficult to get the latest artifact out of circleci.

File: site/content/extend/plugins/community_process.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+---
+title: Plugins on Community Server
+date: 2018-10-01T00:00:00-05:00
+subsection: Plugins (Beta)
+weight: 50
+---
+
+
+Getting your plugin onto community.mattermost.com is a valuable source of feedback. However we must ensure that our community server remains stable for everyone. This document outlines the process of getting your plugin onto the community server. In the future, some of these steps will be required to get your plugin into the integrations marketplace (TBD). When you are ready to have your plugin start this process, ask in the ~Toolkit channel on the community server.
```
Comment (jasonblais) [inline site/content/extend/plugins/community_process.md]: 1 - What's a "community server"?
2 - Who owns the process when someone asks in the ~Toolkit channel?
3 - Who owns bug fixes / maintenance when something breaks?
4 - Is this intended for everyone, including contributor-created plugins?

File: site/content/extend/plugins/community_process.md (hunk 1 above)
Comment (DHaussermann) [reply site/content/extend/plugins/community_process.md]: - Ensure all setup documentation needed is clear and can be successfully followed 
- Functional testing has been done to ensure the integration works as expected
- For Plugins owned by Mattermost - Release testing is added to cover the main functionality of the plugin

File: site/content/extend/plugins/community_process.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+[More info](https://www.apache.org/legal/resolved.html#category-a)
+
+### Complete code review
+
+A more through code review is performed before allowing a plugin on ci-extensions-release. This review works the same as the basic code review, but the developers performing the review will be more through. If the developer that performed the first review is available, they should be one of the reviewers. One of the reviewers should focus their review on the any security implications of the plugin.
+
+### QA pass
+
+QA pass involves getting a memeber of our QA team to take a look and verify the functionality advertised by your plugin.
+
```
Comment (DHaussermann) [inline site/content/extend/plugins/community_process.md]: - Ensure all setup documentation needed is clear and can be successfully followed 
- Functional testing has been done to ensure the integration works as expected
- For Plugins owned by Mattermost - Release testing is added to cover the main functionality of the plugin

**Comment by crspeller** [comment]
@jasonblais I added some clarification to the first paragraph to address some of your points.

@DHaussermann I integrated your steps for QA. 

Ready for re-review.

File: site/content/extend/plugins/community_process.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
+---
+title: Plugins on Community Server
+date: 2018-10-01T00:00:00-05:00
+subsection: Plugins (Beta)
+weight: 50
+---
+
+
+Getting your plugin onto our community server https://community.mattermost.com is a valuable source of feedback. Weather you are a Core Commiter or anyone from the community, we want you to get feedback to improve your plugin. However we must ensure that our community server remains stable for everyone. This document outlines the process of getting your plugin onto the community server. In the future, some of these steps will be required to get your plugin into the integrations marketplace (TBD). When you are ready to have your plugin start this process, ask in the ~Toolkit channel on the community server. The PM, or somone else from the integrations team will help you start the process.
```
Comment (jasonblais) [inline site/content/extend/plugins/community_process.md]:
```suggestion
Getting your plugin onto our community server https://community.mattermost.com is a valuable source of feedback. Whether you are a [Core Committer](https://github.com/mattermost/mattermost-developer-documentation/blob/dae9ecb3b445111479acbaba9f382e9eb263bc52/contribute/getting-started/core-committers) or anyone from the community, we want you to get feedback to improve your plugin. However we must ensure that our community server remains stable for everyone. This document outlines the process of getting your plugin onto the community server. In the future, some of these steps will be required to get your plugin into the integrations marketplace (TBD). When you are ready to have your plugin start this process, ask in the ~Toolkit channel on the community server. The PM, or somone else from the integrations team will help you start the process.
```

**Pull request merged**